File: 1711-find-valid-matrix-given-row-and-column-sums/1711-find-valid-matrix-given-row-and-column-sums.java
class Solution {
    public int[][] restoreMatrix(int[] rowSum, int[] colSum) {
        final int row = rowSum.length;
        final int col = colSum.length;

        int[][] ans = new int[row][col];

        for (int r = 0; r < row; r++) 
            for (int c = 0; c < col; c++) {
                ans[r][c] = Math.min(rowSum[r], colSum[c]);

                rowSum[r] -= ans[r][c];
                colSum[c] -= ans[r][c];
            }

        return ans;
    }
}

File: 1711-find-valid-matrix-given-row-and-column-sums/1711-find-valid-matrix-given-row-and-column-sums.cpp
class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        vector<vector<int>> ans(rowSum.size(), vector<int>(colSum.size()));

        for (int r = 0; r < rowSum.size(); r++) 
            for (int c = 0; c < colSum.size(); c++) {
                ans[r][c] = min(rowSum[r], colSum[c]);

                rowSum[r] -= ans[r][c];
                colSum[c] -= ans[r][c];
            }
        return ans;
    }
};

auto init = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 'c';
}();